src/client/dfs/dfs_sys.c

```diff
@@ -691,6 +691,11 @@ dfs_sys_chmod(dfs_sys_t *dfs_sys, const char *path, mode_t mode)
 	if (rc != 0)
 		return rc;
 
+	if (mode & (S_ISVTX | S_ISGID | S_ISUID)) {
+		D_DEBUG(DB_TRACE, "setuid, setgid, and sticky bit are not supported, and will be ignored.\n");
+		mode &= ~(S_ISVTX | S_ISGID | S_ISUID);
+	}
+
```

Comment:
I have no doubt that this would allow the copy to work however it would also mean the tool was silently doing something other than what it was requested to.

A better approach (to me) would be that this code should just allow the chmod() call, dfs would reject it with ENOTSUP and then the higher level would log this and re-try without the bits set, and could then know it had to modify it's behaviour in order to complete the command and exit non-zero at the end.

Reply:
I have a new PR from a branch in daos-stack, let's move the review there since we'll be able to trigger presubmits on that one.